tuf/pyca_crypto_keys.py

```diff
@@ -0,0 +1,1017 @@
+
+    data:
+      Data (string) used by create_rsa_signature() to generate the signature.
+
+  <Exceptions>
+    tuf.FormatError, if 'private_key' is improperly formatted.
+    
+    TypeError, if 'private_key' is unset.
+
+    tuf.CryptoError, if the signature cannot be generated. 
+
+  <Side Effects>
+    pyca/cryptography's 'RSAPrivateKey.signer()' called to generate the
+    signature.
+
+  <Returns>
+    A (signature, method) tuple, where the signature is a string and the method
+    is 'RSASSA-PSS'.
+  """
+  
+  # Does the arguments have the correct format?
+  # This check will ensure the arguments conform to 'tuf.formats.PEMRSA_SCHEMA'.
+  # and 'tuf.formats.DATA_SCHEMA' 
+  # Raise 'tuf.FormatError' if the checks fail.
+  tuf.formats.PEMRSA_SCHEMA.check_match(private_key)
+  tuf.formats.DATA_SCHEMA.check_match(data) 
+
+  # Signing 'data' requires a private key.  The 'RSASSA-PSS' signing method is
+  # the only method currently supported.
+  method = 'RSASSA-PSS'
+  signature = None
+
+  # Verify the signature, but only if the private key has been set.  The private
+  # key is a NULL string if unset.  Although it may be clearer to explicitly
+  # check that 'private_key' is not '', we can/should check for a value and not
+  # compare identities with the 'is' keyword.  Up to this point 'private_key'
+  # has variable size and can be an empty string.
+  if len(private_key):
+    
+    # pyca/cryptography's expected exceptions when generating RSA key object:
+    # "ValueError/IndexError/TypeError:  When the given key cannot be parsed
+    # (possibly because the passphrase is wrong)."
+    # A passphrase or password is not used to 'private_key', since it should
+    # not be encrypted.
+    try:
+      # 'private_key' (in PEM format) must be converted to a
+      # pyca/cryptography private key object before a signature can be
+      # generated.
+      private_key_object = load_pem_private_key(private_key.encode('utf-8'),
+                                                password=None,
+                                                backend=default_backend())
+   
+      # Calculate the SHA256 hash of 'data' and generate the hash's PKCS1-PSS
+      # signature. 
+      rsa_signer = \
+        private_key_object.signer(padding.PSS(mgf=padding.MGF1(hashes.SHA256()),
+                      salt_length=hashes.SHA256().digest_size), hashes.SHA256())
+                          
+    except ValueError: #pragma: no cover
+      raise tuf.CryptoError('The private key could not be deserialized.')
+ 
+    except cryptography.exceptions.UnsupportedAlgorithm: #pragma: no cover
+      message = 'The private key is encrypted with an unsupported algorithm.'
+      raise tuf.CryptoError(message)
+   
+    # Generate an RSSA-PSS signature.  Raise 'tuf.CryptoError' for the expected
+    # pyca/cryptography exceptions.
+    rsa_signer.update(data)
+    signature = rsa_signer.finalize()
+  
+  else:
+    raise TypeError('The required private key is unset.')
```

Comment:
I would consider this a `ValueError` instead, see also https://github.com/vladimir-v-diaz/tuf/blob/ef2ab091db1aa87d3feec3b0732135d65be3bb16/tuf/pyca_crypto_keys.py#L970-L971